tsfc/driver.py

```diff
@@ -57,6 +55,9 @@ def compile_form(form, prefix="form", parameters=None, interface=None):
     fd = ufl_utils.compute_form_data(form, complex_mode=complex_mode)
     logger.info(GREEN % "compute_form_data finished in %g seconds.", time.time() - cpu_time)
 
+    if interface is None:
+        interface = firedrake_interface.KernelBuilder
```

Comment:
I think it would be slightly nicer to this in `compile_integral` instead (setting the default value of the optional argument to `None`), and let `compile_form` pass through `None` or whatever was specified by the caller.  That way, we only need to tell once (instead of twice) what the default kernel interface is.

Reply:
Good point.